pkg/controller/cluster/agent/shared.go

```diff
@@ -60,13 +62,18 @@ func (s *SharedAgent) Config() ctrlruntimeclient.Object {
 }
 
 func sharedAgentData(cluster *v1alpha1.Cluster, token, nodeName, ip string) string {
+	version := cluster.Spec.Version
+	if cluster.Spec.Version == "" {
+		version = cluster.Status.HostVersion
+	}
```

Comment:
Looking a bit for a different solution I found a very interesting one I didn't know about!

```suggestion
	version := cmp.Or(cluster.Spec.Version, cluster.Status.HostVersion) 
```

Reply:
yeah it seems that they have added that in go 1.22, however I am not sure it is the best use case for our situation, since its just a simple comparison of strings and it really doesnt need a new layer of abstraction